tools/GlobalToolShimGenerator/ShimGenerator.targets

```diff
@@ -0,0 +1,121 @@
+<Project DefaultTarget="GenerateShims">
+  <Import Project="$(MSBuildThisFileDirectory)\ShimGeneratorInner.targets" />
+
+  <Target Name="BuildEmbedAppNameInHostTask">
+    <!--<Message Text="*** BuildEmbedAppNameInHostTask ***" Importance="high"/>-->
+
+    <MSBuild BuildInParallel="False"
+             Targets="Restore;Build"
```

Comment:
Can you ensure this works on a clean machine? It typically doesn't because there are targets that need to be imported after restore and the project needs to be re-evaluated.

Also, any reason this can't just run in the normal restore? I'm concerned we will passing in other important properties for restore.